include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -61,38 +84,38 @@ namespace eve
     constexpr Func<OptionsValues> const& derived() const { return static_cast<Func<OptionsValues>const&>(*this); }
   };
 
-  //====================================================================================================================
-  //====================================================================================================================
+
   template< template<typename> class Func
           , typename OptionsValues
           , typename... Options
           >
-  struct elementwise_callable : callable<Func, OptionsValues, conditional_option, Options...>
+  struct strict_elementwise_callable : callable<Func, OptionsValues, conditional_option, Options...>
   {
     using func_t =  Func<OptionsValues>;
 
     struct ignore { template<typename T> operator T() { return T{}; } };
 
     template<callable_options O, typename T, typename... Ts>
-    EVE_FORCEINLINE constexpr auto behavior(auto arch, O const& opts, T x0,  Ts const&... xs) const
-    requires(match_option<condition_key,O,ignore_none_>)
+    constexpr EVE_FORCEINLINE auto adapt_call(auto a, O const& o, T x,  Ts const&... xs) const
     {
-      constexpr bool supports_call = requires{ func_t::deferred_call(arch, opts, x0, xs...); };
-      constexpr bool any_simd     = (simd_value<T> || ... || simd_value<Ts>);
+      constexpr bool has_implementation = requires{ func_t::deferred_call(a, o, x, xs...); };
+      constexpr bool is_emulated        = requires{ map(this->derived(), x, xs...); };
+      constexpr bool any_emulated       = (has_emulated_abi_v<T> || ... || has_emulated_abi_v<Ts>);
+      constexpr bool any_aggregated     = (has_aggregated_abi_v<T> || ... || has_aggregated_abi_v<Ts>);
 
-      // If a deferred call is present, let's call it
-      if      constexpr( supports_call )  return func_t::deferred_call(arch, opts, x0, xs...);
-      else if constexpr( any_simd )
-      {
-        // if not, try to slice/aggregate from smaller wides or to map the scalar version
-        if constexpr((has_aggregated_abi_v<Ts> || ...)) return aggregate(this->derived(), x0, xs...);
-        else                                            return map(this->derived(), x0, xs...);
-      }
-      else
-      {
-        static_assert(any_simd, "[EVE] - Missing scalar implementation for current callable");
-        return ignore{};
-      }
+      if      constexpr(any_aggregated)               return aggregate(this->derived(), x, xs...);
+      else if constexpr(any_emulated && is_emulated ) return map(this->derived(), x, xs...);
```

Comment:
Not having 'is_emulated' be true is a bug in eve, right?

so !is_emulated should report a bug here and not fall back